### DISCUSSION TITLE
Astrocartography

### DISCUSSION THREAD

**Discussion by stoow84** [community/community#61882] [Programming Help]
### Select Topic Area

Question

### Body

Hi, I am new to coding and am trying to build a map of the Earth which i can overlay planetary positions. Are there any projects of similar topic that i can view for ideas on where to start this project. Any help is greatly appreciated.

Similar to the image but preferably in a globe view/sphere which can be rotated and zoomed.
![image](https://github.com/community/community/assets/85204742/a6ea192a-8372-4b5a-b077-8d64db65564c)

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @stoow84 - this looks super interesting, I've dabbled a bit in astrocartography myself! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Reply by wizardofoz369** [reply]
Can you offer any suggestions on python packages to utilise to begin building this type of code - astrocartography.

**Comment by hokkar** [comment]
I build a new one [https://astrokartographie-analyse.de/](https://astrokartographie-analyse.de/)

**Comment by OgabekHub** [comment]
Great project idea! Here are some resources to get you started:

**Similar open-source projects:**
- `d3-celestial` — planet/star positions on an interactive map
- `Stellarium Web` — open source, runs in browser
- `CesiumJS` — 3D globe with overlays (very powerful)

**For a rotatable 3D globe:**
- `Three.js` + a globe texture — most flexible option
- `Globe.gl` — easiest to start with, built on Three.js
- `D3.js` + GeoJSON — better for 2D flat map projections

**Planetary position data:**
- `astronomia` (npm) — calculates planet positions in JS
- NASA JPL Horizons API — free, accurate ephemeris data

**Recommended starting point:**
Globe.gl for the 3D sphere + astronomia for planet 
coordinates. Both have good documentation for beginners.
